ckanext/datastore/backend/postgres.py

```diff
@@ -2238,15 +2392,12 @@ def delete(self, context: Context, data_dict: dict[str, Any]):
             #FIXME: flagged issue, referencing someone else's table will limit their table from the constraints.
             except (IntegrityError, ProgrammingError, InternalError) as e:
                 if e.orig.pgcode == _PG_ERR_CODE['row_referenced_constraint'] or e.orig.pgcode == _PG_ERR_CODE['table_referenced_constraint']:
-                    # FIXME: better error message??
-                    raise ValidationError({
-                        'foreign_constraints': ['Cannot delete records or table'
-                                                ' because of a reference to another table.'],
-                        'info': {
-                            'orig': str(e.orig),
-                            'pgcode': e.orig.pgcode
-                        }
-                    })
+                    # (canada fork only): parse constraint sql errors
+                    # TODO: upstream contrib!!
+                    errmsg = 'Cannot delete records or table because of '\
+                             'a reference to another table. '\
+                             'Referencing {refKeys}({refValues}) from {refTable}.'
```

Comment:
a hack like this allows the messages to be picked up by babel with extract_messages without actually calling gettext.
```suggestion
                    _ = lambda x:x
                    errmsg = _(
                        'Cannot delete records or table because of '
                        'a reference to another table. '
                        'Referencing {refKeys}({refValues}) from {refTable}.')
```

Reply:
oooo right, forgot these puppies have to be picked up.